crates/file_store/src/store.rs

```diff
@@ -63,17 +55,24 @@ where
         })
     }
 
-    /// Open an existing [`Store`].
+    /// Load an existing [`Store`].
     ///
-    /// Use [`create_new`] to create a new `Store`.
+    /// Use [`create`] to create a new [`Store`].
     ///
     /// # Errors
     ///
-    /// If the prefixed bytes of the opened file does not match the provided `magic`, the
-    /// [`FileError::InvalidMagicBytes`] error variant will be returned.
+    /// If the prefixed bytes of the loaded file does not match the provided `magic`, a
+    /// [`StoreErrorWithDump`] will be returned with the [`StoreError::InvalidMagicBytes`] error variant in
+    /// its error field and changeset field set to [`Option::None`]
     ///
-    /// [`create_new`]: Store::create_new
-    pub fn open<P>(magic: &[u8], file_path: P) -> Result<Self, FileError>
+    /// If there exist changesets in the file, [`load`] will try to aggregate them in
+    /// a single changeset to verify their integrity. If aggregation fails
+    /// [`StoreErrorWithDump`] will be returned with the [`StoreError::Bincode`] error variant in
+    /// its error field and the aggregated changeset so far in the changeset field.
+    ///
+    /// [`create`]: Store::create
+    /// [`load`]: Store::load
+    pub fn load<P>(magic: &[u8], file_path: P) -> Result<(Option<C>, Self), StoreErrorWithDump<C>>
```

Comment:
I have a slight preference for returning `(Self, Option<C>)`, i.e. make `Self` the first item in the tuple, but I don't have a good reason other than it feels more ergonomic.

Reply:
Fixed by 5a7fe4259c7a7c62b24973acdcd1961c46d69bdd